src/main/java/com/rtb/projectmanagementtool/task/TaskController.java

```diff
@@ -71,6 +71,38 @@ public void removeUser(TaskData task, long userID) {
     }
   }
 
+  public void setComplete(long taskID) {
+    setComplete(getTaskByID(taskID));
+  }
+
+  public void setComplete(TaskData task) {
+    ArrayList<TaskData> subtasks = getSubtasks(task);
+    for (TaskData subtask : subtasks) {
+      if (subtask.getStatus() != Status.COMPLETE) {
```

Comment:
This validates if all subtasks are completed before setting the complete flag?  I think this is a good idea.  I would suggest splitting this up into two methods though.  There could be cases where you'd want to just validate the subtasks without actually changing the parent  task (maybe for changing a UI element or something).  There could also be scenarios where you'd need to change the parent task status blindly without looking at the subtasks first.

ie:
```
if (!allSubtasksCompleted(task) {
   return
}

setComplete(task);
```